Automattic/wp-calypso
# Jetpack Backup: enable download and backup browser without failing Jetpack connection

Resolves https://github.com/Automattic/jetpack-backup-team/issues/468

## Proposed Changes

* On download and backup contents pages routes:
  * Remove `showUnavailableForVaultPressSites` callback
    * If a site has a legacy VaultPress plan, they will not be able to access Jetpack Cloud anyway, so this is no longer needed.
  * Remove `showJetpackIsDisconnected` callback
    * In case a site is still connected to Jetpack, but it is not active (e.g. site is down), we should let users download and view their backup contents. 

## Testing Instructions



* Spin up a Calypso or Jetpack Cloud live branch
* Pick a site with Jetpack VaultPress Backup plan and failing Jetpack connection (protip: if you have an old Jurassic Ninja site, it might work)
* Navigate to Activity Log
* Pick a backup and click on `Actions` -> `Download backup`
* Try to download a backup and ensure it works okay
* Now try again in the Activity Log, but this time navigating to `Actions` -> `View files`
* Try downloading a single file or a full backup and ensure it works okay.
* You can also try to restore your site and it should display `Jetpack connection has failed` view.

## Pre-merge Checklist
